executor/src/witgen/vm_processor.rs

```diff
@@ -460,6 +441,57 @@ impl<'a, T: FieldElement> VmProcessor<'a, T> {
         }
     }
 
+    /// Given a row and identity index, computes any updates, applies them and returns
+    /// whether any progress was made.
+    pub fn process_identity<Q: QueryCallback<T>>(
+        &mut self,
+        row_index: DegreeType,
+        identity: &'a Identity<Expression<T>>,
+        mutable_state: &mut MutableState<'a, '_, T, Q>,
+        unknown_strategy: UnknownStrategy,
+    ) -> Result<(bool, bool), EvalError<T>> {
+        // Create row pair
+        let global_row_index = self.row_offset + row_index;
+        let row_pair = RowPair::new(
+            &self.data[row_index as usize],
+            &self.data[row_index as usize + 1],
+            global_row_index,
+            self.fixed_data,
+            unknown_strategy,
+        );
+
+        // Compute updates
```

Comment:
Where does this code come from?

Reply:
This was just an intermediate step. I think it was copied from `Processor`; now it uses `Processor instead,`